### PR TITLE
awful.tag: Add `gap_count` property

### PR DESCRIPTION
**Pull request by Elv13**
This restore a feature removed during the API cleanup.

This still need a better name

### CODE REVIEW

**Comment by Elv13** [comment]
@zainin @actionless This is totally untested and will require a better name please test.

> and what do you think about also checking awful.tag.getmfpol(t) == "expand" ?

@actionless Can you elaborate? You mean being able to disable the gaps differently when the mfpol is set? In that case, wont that also affect the "vertical" spacing in an unwanted way?

File: lib/awful/tag.lua [hunk 1]
```diff
@@ -795,6 +797,32 @@ function tag.incgap(add, t)
...
+-- @beautiful beautiful.useless_gap_count
+-- @param integer (default: 0)
+-- @see gap
+-- @see gap_count
+
+--- The minimum number of client at which to enable gaps.
+--
+-- **Signal:**
+--
+-- * *property::gap\_count*
```
Comment (psychon) [inline lib/awful/tag.lua]: Isn't it `property::useless_gap_count` due to using a different name internally?

File: lib/awful/tag.lua [hunk 2]
```diff
@@ -55,6 +65,39 @@ end
 
 --- The number of elements kept in the history.
 -- @tfield integer awful.tag.history.limit
+-- @tparam[opt=20] integer limit
+
+--- The gap between clients (in points).
+-- @tfield[opt=0] integer awful.tag.defaults.gap
+-- @tparam[opt=0] integer gap
+-- @see gap
+-- @see gap_count
```
Comment (psychon) [inline lib/awful/tag.lua]: Can't you move these docs directly to the variable definition above?
```
--- Foo is a great value
-- @field[opt=42] awful.tag.defaults.foo
tag.foo = 42
```

**Comment by psychon** [comment]
Name... uhm... this does `minimum_clients_for_useless_gaps`, but that's quite a long name. When does anyone want to set this to something else than 0 or 1? (Actually, the implementation uses `>`, not `>=`, so the only interesting values would be -1 and 0, but -1 cannot be set, right?) If this were turned into a boolean, it could be `useless_gaps_ignore_single_client` or something like that. Some more random rambling: `useless_gaps_require_multiple_clients`, `useless_gaps_for_single_client` (this one being an integer that is used when just one client is visible).... hm, this is hard

**Comment by actionless** [comment]
@Elv13 
> Can you elaborate? 
see example here: https://github.com/awesomeWM/awesome/issues/826 or mb better copy it to the config and see it in action -- the idea is what when master_fill_policy is set to mwfact when window have big gaps on the sides, and since globally gaps are enabled it looks a bit illogical what that single window will have only side gaps but not upper/bottom small ones

**Comment by zainin** [comment]
@Elv13 
Works fine, I've found no issues.

@psychon 
A boolean sounds like the best choice. Like you've said, it's unlikely someone would want >1 minimum clients.

**Comment by Elv13** [comment]
Obseleted by #1066

**Pull request closed without merging**